Library/Homebrew/os/linux/elf.rb

```diff
@@ -172,6 +200,14 @@ def needed_libraries_using_readelf(path)
     end
   end
 
+  def patchelf_patcher
+    @patchelf_patcher ||= begin
```

Comment:
```suggestion
  def patchelf_patcher
    return unless HOMEBREW_PATCHELF_RB
    
    @patchelf_patcher ||= begin
```

Reply:
@MikeMcQuaid if this is for development time safety, would it not be better to exit with a helpful message?

Reply:
I feel like returning `nil` is more fitting with the other methods in this file.

Reply:
would exiting not be easier to debug then returning nil ? 
Also it would make the fact explicit that we are not supposed to call this method when HOMEBREW_PATCHELF_RB is not defined.

Reply:
Again: I think it's worth being consistent with the other code in this method.

Reply:
@MikeMcQuaid  applied the requested changes.

It puzzles me on the question of trading easy debugging and lesser code for the sake of consistency. 
Can you share the reasoning so i "get it".

Reply:
Thanks!



> It puzzles me on the question of trading easy debugging and lesser code for the sake of consistency.

I think consistency does easy debugging. In this case, I'm not sure how an error would be much easier than `nil` (when it's a method call, anyway).

Reply:
> I think consistency does easy debugging
no doubt.

> In this case, I'm not sure how an error would be much easier than nil (when it's a method call, anyway).

return `[nil, []]` implicitly means, something went wrong  that's okay  we'll  continue with defaults anyway.

my main concern is  it won't hint the programmer they are not supposed to 
invoke this method when `HOMEBREW_PATCHELF_RB` is not set.